src/canvasPanelComponents/slideshow/src/components/Slideshow/SlideShow.tsx

```diff
@@ -0,0 +1,257 @@
+import React, { useState, useEffect, useRef } from 'react';
+import {
+  Manifest,
+  Fullscreen,
+  RangeNavigationProvider,
+  withBemClass,
+  Responsive,
+  // @ts-ignore
+} from '@canvas-panel/core';
+import MobileViewer from '../MobileViewer/MobileViewer';
+import TapDetector from '../TapDetector/TapDetector';
+import SimpleSlideTransition from '../SimpleSlideTransition/SimpleSlideTransition';
+import ProgressIndicator from '../ProgressIndicator/ProgressIndicator';
+import Slide from '../Slide/Slide';
+// @ts-ignore
+import CanvasNavigation from '../CanvasNavigation/CanvasNavigation.tsx';
+import PeekComponent from '../PeekComponent/PeekComponent';
+
+import './SlideShow.scss';
+
+interface SlideShowProps {
+  mobileBreakPoint: number;
+  backgroundColor: string;
+  addressable: boolean;
+  id: number;
+  jsonLd: object;
+  manifestUri: string;
+  renderPanel: () => void;
+  bem: any;
+}
+
+const SlideShow: React.FC<SlideShowProps> = ({
+  mobileBreakPoint = 767,
+  backgroundColor = '#000000',
+  addressable = false,
+  id,
+  jsonLd,
+  manifestUri,
+  renderPanel,
+  bem,
+}) => {
+  const [innerWidth, setInnerWidth] = useState(0);
+  const [qualifiesForMobile, setQualifiesForMobile] = useState(false);
+  const [isMobileFullScreen, setIsMobileFullScreen] = useState(false);
+  const [down, setDown] = useState(false);
+  const [open, setOpen] = useState(false);
+  const [offset, setOffset] = useState(0);
+  const [inFocus, setInFocus] = useState(false);
+  let touchDetector: any | null = null;
+  let viewport: any;
+  const slideshowEl = useRef(null);
+
+  useEffect(() => {
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    touchDetector = null;
+  }, [down]);
+
+  useEffect(() => {
+    if (touchDetector !== null) {
+      touchDetector.onTap(onTap);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [touchDetector]);
+
+  useEffect(() => {
+    window.addEventListener('resize', () => setInnerWidth(window.innerWidth));
+    return () =>
+      window.removeEventListener('resize', () =>
+        setInnerWidth(window.innerWidth)
+      );
+  }, []);
+
+  useEffect(() => {
+    setQualifiesForMobile(window.innerWidth <= mobileBreakPoint);
+  }, [innerWidth, mobileBreakPoint]);
+
+  const nextInRange = (fromHOC: () => void) => {
+    if (viewport) viewport.viewer.viewer.viewport.applyConstraints(true);
+    fromHOC();
+  };
+
+  const previousInRange = (fromHOC: () => void) => {
+    if (viewport) viewport.viewer.viewer.viewport.applyConstraints(true);
+    fromHOC();
+  };
```

Comment:
This is one of the big problems working with the current API. I was hoping to get `use-open-seadragon` included at some point, which would let us do something like this: https://github.com/stephenwf/use-open-seadragon/blob/master/src/example.tsx#L9 to get the current viewer. Nothing we can do right now, but it will be easier in the future.